pageserver/src/tenant.rs

```diff
@@ -3229,6 +3230,30 @@ impl Tenant {
     pub fn cached_synthetic_size(&self) -> u64 {
         self.cached_synthetic_tenant_size.load(Ordering::Relaxed)
     }
+
+    /// Flush any in-progress layers, schedule uploads, and wait for uploads to complete.
+    ///
+    /// This function can take a long time: callers should wrap it in a timeout if calling
+    /// from an external API handler.
+    pub async fn flush_remote(&self) -> anyhow::Result<()> {
+        let timelines = self.timelines.lock().unwrap().clone();
+
+        for (timeline_id, timeline) in timelines {
+            tracing::info!(%timeline_id, "Flushing...");
+            timeline.freeze_and_flush().await?;
+            tracing::info!(%timeline_id, "Waiting for uploads...");
+            if let Some(client) = &timeline.remote_client {
+                client.wait_completion().await?;
+            }
+        }
+
+        match self.deletion_queue_client.flush_execute().await {
+            Ok(_) => {}
+            Err(DeletionQueueError::ShuttingDown) => {}
+        }
```

Comment:
Is this included because of the name "flush" to mean all of the possible flushes or is something expected to be removed now...? I can't see what could it be, but perhaps this needs to be in the joinset as well as layer flush should not add deletions right away, compaction might.

Reply:
I've added a comment to explain why this is done (it's for flushing any deletions that might have been enqueued before we entered flush_ms)

Reply:
Explanation sounds reasonable, but why not put this into the joinset as well?